### PR TITLE
BUG: fix bug where mad ignores center if center is not callable

### LINKED ISSUE #6999: BUG: issue with how mad handles non-callable center arguments
#### Describe the bug

The center argument in the mad function is supposed to set the location estimate around which the mad will be computed. If center is not callable, the the function sets center to 0, and ignores the input, but the documentation states that center can also be a float. This is what the docs say mad should return: median(abs(a - center))/c. This was already mentioned in #6969. One solution is to change the docs, and add an assert that center is callable. Another possibility is to fix the line that sets center to zero if center is not callable, assert that center is float in that case and use the passed value in the computation. Either way, I'd be happy to write a PR fixing it.

#### Code Sample, a copy-pastable example if possible

```python
import numpy as np
from statsmodels.robust.scale import mad
a = np.array([0, 1, 2])

mad(a, center=-10) #1.482602218505602
mad(a, center=0) #1.482602218505602
np.median(np.abs(a - (-10))) / .6745 #16.30837657524092, this is the right answer for mad(a, center=-10)
```
<details>

**Note**: As you can see, there are many issues on our GitHub tracker, so it is very possible that your issue has been posted before. Please check first before submitting so that we do not have to handle and close duplicates.

**Note**: Please be sure you are using the latest released version of `statsmodels`, or a recent build of `master`. If your problem has been fixed in an unreleased version, you might be able to use `master` until a new release occurs. 

**Note**: If you are using a released version, have you verified that the bug exists in the master branch of this repository? It helps the limited resources if we know problems exist in the current master so that they do not need to check whether the code sample produces a bug in the next release.

</details>

If the issue has not been resolved, please file it in the issue tracker.

#### Expected Output

The output of mad(a, center=center) should be equal to np.median(np.abs(a - center)) / .6745.
#### Output of ``import statsmodels.api as sm; sm.show_versions()``

<details>

INSTALLED VERSIONS
------------------
Python: 3.8.2.final.0
OS: Linux 5.4.0-42-generic #46-Ubuntu SMP Fri Jul 10 00:24:02 UTC 2020 x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8

statsmodels
===========

Installed: v0.12.0rc0+31.g1e6b7c6ef (/home/esmucler/Packages/statsmodels/statsmodels)

Required Dependencies
=====================
cython: 0.29.21 (/home/esmucler/Packages/statsmodels/venv/lib/python3.8/site-packages/Cython)
numpy: 1.19.1 (/home/esmucler/Packages/statsmodels/venv/lib/python3.8/site-packages/numpy)
scipy: 1.5.2 (/home/esmucler/Packages/statsmodels/venv/lib/python3.8/site-packages/scipy)
pandas: 1.1.0 (/home/esmucler/Packages/statsmodels/venv/lib/python3.8/site-packages/pandas)
    dateutil: 2.8.1 (/home/esmucler/Packages/statsmodels/venv/lib/python3.8/site-packages/dateutil)
patsy: 0.5.1 (/home/esmucler/Packages/statsmodels/venv/lib/python3.8/site-packages/patsy)

Optional Dependencies
=====================
matplotlib: 3.3.1 (/home/esmucler/Packages/statsmodels/venv/lib/python3.8/site-packages/matplotlib)
    backend: agg 
cvxopt: Not installed
joblib: Not installed

Developer Tools
================
IPython: 7.17.0 (/home/esmucler/Packages/statsmodels/venv/lib/python3.8/site-packages/IPython)
    jinja2: 2.11.2 (/home/esmucler/Packages/statsmodels/venv/lib/python3.8/site-packages/jinja2)
sphinx: 3.2.1 (/home/esmucler/Packages/statsmodels/venv/lib/python3.8/site-packages/sphinx)
    pygments: 2.6.1 (/home/esmucler/Packages/statsmodels/venv/lib/python3.8/site-packages/pygments)
pytest: 6.0.1 (/home/esmucler/Packages/statsmodels/venv/lib/python3.8/site-packages/pytest)
virtualenv: Not installed

</details>

### PR DESCRIPTION
**Pull request by esmucler**
- [X] closes #6999
- [X] tests added / passed. 
- [X] code/documentation is well formatted.  
- [X] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy-1.15.1/dev/gitwash/development_workflow.html#writing-the-commit-message). 

<details>

**Notes**:

* It is essential that you add a test when making code changes. Tests are not 
  needed for doc changes.
* When adding a new function, test values should usually be verified in another package (e.g., R/SAS/Stata).
* When fixing a bug, you must add a test that would produce the bug in master and
  then show that it is fixed with the new code.
* New code additions must be well formatted. Changes should pass flake8. If on Linux or OSX, you can
  verify you changes are well formatted by running 
  ```
  git diff upstream/master -u -- "*.py" | flake8 --diff --isolated
  ```
  assuming `flake8` is installed. This command is also available on Windows 
  using the Windows System for Linux once `flake8` is installed in the 
  local Linux environment. While passing this test is not required, it is good practice and it help 
  improve code quality in `statsmodels`.
* Docstring additions must render correctly, including escapes and LaTeX.

</details>

### CODE REVIEW

File: statsmodels/robust/scale.py [hunk 1]
```diff
@@ -45,8 +45,11 @@ def mad(a, c=Gaussian.ppf(3/4.), axis=0, center=np.median):
...
     c = float_like(c, 'c')
-    if callable(center) and a.size:
-        center = np.apply_over_axes(center, a, axis)
+    if a.size:
+        if callable(center):
+            center = np.apply_over_axes(center, a, axis)
+        else:
+            center = float_like(center, 'center')
     else:
         center = 0.0
```
Comment (josef-pkt) [inline statsmodels/robust/scale.py:54]: What case does this handle?
AFAICS, not a.size would only check for empty arrays.


```
np.median(np.abs(np.array([]) - 0))
nan
```

Comment (bashtage) [reply statsmodels/robust/scale.py:54]: I think you just need an `elif center is None` and then set center to 0, else cast check to float.

**Review by bashtage — CHANGES_REQUESTED** [review]
Small changes needed

**Comment by esmucler** [comment]
> Small changes needed
" I think you just need an elif center is None and then set center to 0, else cast check to float."
If center is None, wouldn't it be better to raise an error? It seems more likely that the user passed center as None erroneously than meaning to use 0 as a center. This is what happens now if center is None, float_like throws a TypeError

File: statsmodels/robust/scale.py (hunk 1 above)
Comment (esmucler) [reply statsmodels/robust/scale.py:54]: If a is empty and I don't set center to something then the final line throws a ZeroDivisionError when running line 76 of test_scale.py, exactly the line that tests that mad returns nan when a is empty.

**Comment by bashtage** [comment]
> Small changes needed
> " I think you just need an elif center is None and then set center to 0, else cast check to float."
> If center is None, wouldn't it be better to raise an error? It seems more likely that the user passed center as None erroneously than meaning to use 0 as a center. This is what happens now if center is None, float_like throws a TypeError

Yes, you are right.  I think the simplest is 

```
if not a.size:
    # center is not relevant when a is empty
    center = 0.0
elif callable(center):
    center = np.apply_over_axes(center, a, axis)
else:
    center = float_like(center, "center")
```

**Comment by esmucler** [comment]
> > Small changes needed
> > " I think you just need an elif center is None and then set center to 0, else cast check to float."
> > If center is None, wouldn't it be better to raise an error? It seems more likely that the user passed center as None erroneously than meaning to use 0 as a center. This is what happens now if center is None, float_like throws a TypeError
> 
> Yes, you are right. I think the simplest is
> 
> ```
> if not a.size:
>     # center is not relevant when a is empty
>     center = 0.0
> elif callable(center):
>     center = np.apply_over_axes(center, a, axis)
> else:
>     center = float_like(center, "center")
> ```

Yep, this is better, done in b3e76b98cbd2b523e5b698acf2f5557fdb23ae01

**Comment by bashtage** [comment]
Thanks!

**Pull request merged**